app/jobs/workoff_arbiter.rb

```diff
@@ -61,12 +61,40 @@ def add_worker!
       ],
     }
 
-    ecs.run_task(payload)
-    @notifier.ping("Added a workoff worker. Metric was #{metric.round} (#{_dj_scope.pluck(:id).count} jobs enqueued) with #{_current_worker_count} workers right now (this might include the just-created one).")
+    _num_workers_to_add.times do
+      ecs.run_task(payload)
+      @notifier.ping("Added a workoff worker. Long jobs pending: #{_long_jobs.count}. Metric was #{metric.round} (#{_dj_scope.pluck(:id).count} jobs enqueued) with #{_current_worker_count} workers right now (this might include the just-created one).")
+    end
   end
 
   private
 
+  def _long_work_pending?
+    _long_jobs.any?
+  end
+
+  def _num_workers_to_add
+    if _long_work_pending?
+      [_long_jobs.count, 3].min
+    else
+      1
```

Comment:
Is this going to mean to mean it adds a worker every time, or am I missing something? I read this as if there are no long jobs pending, add a worker.

Reply:
the add worker method only runs when needs_worker? is true, so it's more of a default of adding one _when we are going to add_